recursion.rb

```diff
@@ -1,16 +1,54 @@
 
 def fact(n)
+  # base cases
+  # 1! = 1
+  # 0! = 1
+  # recursive case
+  # n! = n * (n-1)!
+  if n == 0 || n ==1
+    return 1
+  end
+  return n * fact(n-1)
 end
 
 def fib(n)
+  if n == 1 || n == 2
+    return 1
+  end
+  return fib(n-1) + fib(n-2)
 end
 
-def pal(s)
+def binary(n)
+  binary_helper(n, "", "")
 end
 
-def binary(n)
+def binary_helper(digits, binary, result)
+  if digits == 0
+    result += binary + " "
+  else
+    result = binary_helper(digits-1, binary+"0", result)
+    result = binary_helper(digits-1, binary+"1", result)
+  end
+  return result
 end
 
+def pal(word)
+  first = word[0]
+  last = word[-1]
+  length = word.length
+
+  #base
+  return false if first != last
+  return true if length <= 1
+
+  #recursion
+  word.slice!(0)
+  word.slice!(-1)
+
+  pal(word)
+end
```

Comment:
Good solution! The named variables first/last/length make this solution particularly easy to follow. Good style.